ambry-cloud/src/main/java/com/github/ambry/cloud/HelixVcrClusterParticipant.java

```diff
@@ -105,21 +114,37 @@ public HelixVcrClusterParticipant(CloudConfig cloudConfig, ClusterMapConfig clus
    * @param partitionIdStr The partitionIdStr notified by Helix.
    */
   public void addPartition(String partitionIdStr) {
-    PartitionId partitionId = partitionIdMap.get(partitionIdStr);
+    // Add partition with retry logic in case Ambry clustermap is not update to date.
+    retryExecutor.runWithRetries(retryPolicy, callback -> doAddPartition(partitionIdStr, callback), e -> true,
+        (result, exception) -> {
+          if (exception != null) {
```

Comment:
If we try to handle only one type of exception in retries -- then it should say so.

Reply:
?